src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -239,4 +243,55 @@ public static Exam parseExam(String name, String date, String startTime, String
                 : Optional.empty();
         return new Exam(examName, modBookDate, timeslot, venue, link);
     }
+
+    /**
+     * Parses a {@code String lessonName} into a {@code LessonName}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code lessonName} is invalid.
+     */
+    public static LessonName parseLessonName(String lessonName) throws ParseException {
+        requireNonNull(lessonName);
+        String trimmedLessonName = lessonName.trim();
+        if (!LessonName.isValidLessonName(trimmedLessonName)) {
+            throw new ParseException(LessonName.MESSAGE_CONSTRAINTS);
+        }
+        return new LessonName(trimmedLessonName);
+    }
+
+    /**
+     * Parses a {@code String dayString} into a {@code Day}
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code dayString} is invalid.
+     */
+    public static Day parseDay(String dayString) throws ParseException {
+        requireNonNull(dayString);
+        String trimmedDay = dayString.trim();
+        if (!Day.isValidDay(trimmedDay)) {
+            throw new ParseException(Day.MESSAGE_CONSTRAINTS);
+        }
+        return Day.valueOf(trimmedDay);
+    }
+
+    /**
+     * Parses a {@code String name}, {@code String dayString}, {@code String startTime}, {@code String endTime},
+     * {@code Optional<String> venue}, {@code Optional<String> link} into an {@code Lesson}
+     *
+     * @throws ParseException if any of the data strings are invalid.
+     */
+    public static Lesson parseLesson(String name, String dayString, String startTime, String endTime,
+                                     Optional<String> venueName, Optional<String> linkString) throws ParseException {
+        LessonName lessonName = parseLessonName(name);
+        Day day = parseDay(dayString);
+        Timeslot timeslot = parseTimeslot(startTime, endTime);
+
+        Optional<Venue> venue = venueName.isPresent()
+                ? Optional.of(parseVenue(venueName.get()))
+                : Optional.empty();
+        Optional<Link> link = linkString.isPresent()
+                ? Optional.of(parseLink(linkString.get()))
+                : Optional.empty();
```

Comment:
can use `map` here instead of breaking optional abstraction

Reply:
rejected.